src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/needbraces/InputNeedBracesCaseDefaultNoSingleLine.java

```diff
@@ -0,0 +1,40 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks.needbraces;
+
+public class InputNeedBracesCaseDefaultNoSingleLine {
```

Comment:
please define Check configuration in block comment above class.
Reminder: Input should serve single Check config. Lines with violations should be marked as trailing comments.
with such model you we will a archive easy to validate behavior, almost without reading UT. It will looks like you run checkstyle on single Check config in IDE and it highlights results. 

unfortunately existing Input files might be used by few UT methods, with different config. I hope one day we will resolve this too.

Reply:
done